packages/base-driver/lib/basedriver/extension-core.ts

```diff
@@ -0,0 +1,128 @@
+import {logger} from '@appium/support';
+import {EventEmitter} from 'node:events';
+import type {
+  AppiumLogger,
+  BidiModuleMap,
+  BiDiResultData,
+  StringRecord,
+} from '@appium/types';
+import {
+  MAX_LOG_BODY_LENGTH,
+} from '../constants';
+import {errors} from '../protocol';
+import {BIDI_COMMANDS} from '../protocol/bidi-commands';
+import _ from 'lodash';
+import {generateDriverLogPrefix} from './helpers';
+
+export class ExtensionCore {
+  bidiEventSubs: Record<string, string[]>;
+  bidiCommands: BidiModuleMap = BIDI_COMMANDS as BidiModuleMap;
+  _logPrefix?: string;
+  protected _log: AppiumLogger;
+  // used to handle driver events
+  eventEmitter: NodeJS.EventEmitter;
```

Comment:
consider using readonly for properties that are not going to be reassigned  during instance lifecycle